### DISCUSSION TITLE
Whether Next.js recommend to use `react-router-dom` now?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [vercel/next.js#50775] [Help] [answered]
### Summary

Whether Next.js recommend to use `react-router-dom` now? 

we know next.js have `Pages Router` and `App Router` two ways for routing. 
os today whether there need `react-router-dom`  anymore?  

and if not, which version don't recommend it?

**Comment by devjiwonchoi** [comment, marked as answer]
Either `Page Router` and `App Router` are not **required** to use `react-router-dom` since they have their own `Router` system based on the `folder structure`.

Also, in the `App Router`, the `useRouter()` function differs from the `Page Router`.

> The new useRouter hook should be imported from next/navigation and not next/router

Therefore it is recommended to use the `Link` component for navigation rather than `router.push`

> Recommendation: Use the [<Link> component](https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#link-component) for navigation unless you have a specific requirement for using useRouter.

If you need to use `react-router-dom` in a specific case, please share it so that I can provide more options for you.

## Reference
- Page Router: https://nextjs.org/docs/pages/building-your-application/routing
- App Router: https://nextjs.org/docs/app/building-your-application/routing
- App Router `useRouter`: https://nextjs.org/docs/app/api-reference/functions/use-router
